src/oatpp/network/Server.cpp

```diff
@@ -45,28 +46,48 @@ Server::Server(const std::shared_ptr<ServerConnectionProvider> &connectionProvid
 void Server::mainLoop(Server *instance) {
```

Comment:
It might be better to split this into 
- loop with condition
- loop without condition

Reply:
Ha, its funny because I originally just did that and scraped it later on to "not bloat things up".  When the loop logic grew a bit I did not split it up again. Just wanted to hear what you think about this new functionality before optimising things.